### PR TITLE
Enhance pod-autoscaler experiment by adding abort chaos

### PR DESCRIPTION
**Pull request by adhaamehab**
Fixes [2145](https://github.com/litmuschaos/litmus/issues/2145)

Signed-off-by: adhaamehab <adhaamehab.me@gmail.com>

### CODE REVIEW

File: experiments/generic/pod-autoscaler/pod-autoscaler.go [hunk 1]
```diff
@@ -71,6 +72,10 @@ func main() {
 		"Ramp Time": experimentsDetails.RampTime,
 	})
 
+	// Calling AbortWatcher go routine, it will continuously watch for the abort signal for the entire chaos duration and generate the required events and result
+	// It is being invoked here, as opposed to within the chaoslib, as these experiments do not need additional recovery/chaos revert steps like in case of network experiments
```
Comment (uditgaurav) [inline experiments/generic/pod-autoscaler/pod-autoscaler.go:76]: @adhaamehab  For this experiment, when we abort the chaos which has already scaled the app pod (say to 10 replicas from 3 replicas) the recovery step should bring it back to the original replicas (which is 3).

Comment (ispeakc0de) [reply experiments/generic/pod-autoscaler/pod-autoscaler.go:76]: You can add the requested changes [here](https://github.com/litmuschaos/litmus-go/blob/master/chaoslib/litmus/pod-autoscaler/lib/pod-autoscaler.go#L98) and take the reference from [this](https://github.com/litmuschaos/litmus-go/blob/master/chaoslib/litmus/network-chaos/helper/netem.go#L88) [88-124]

**Comment by ispeakc0de** [comment]
Hi @adhaamehab 

If you have any doubt/confusion or problem statement is not clear. Please feel free to ask.

**Comment by uditgaurav** [comment]
The changes have been addressed with https://github.com/litmuschaos/litmus-go/pull/185 and the issue is resolved now. So closing it. Feel free to open a new PR for further enhancement.

**Pull request closed without merging**